File: app/src/main/java/com/pratik/moviesapp/models/Movie.java
package com.pratik.moviesapp.models;


public class Movie {

    private Results[] results;

    private String page;

    private String total_pages;

    private String total_results;

    public Movie(Results[] results, String page, String total_pages, String total_results) {
        this.results = results;
        this.page = page;
        this.total_pages = total_pages;
        this.total_results = total_results;
    }

    public Results[] getResults ()
    {
        return results;
    }

    public String getPage() {
        return page;
    }

    public String getTotal_pages() {
        return total_pages;
    }

    public String getTotal_results() {
        return total_results;
    }

    public void setResults (Results[] results)
    {
        this.results = results;
    }


    public void setPage (String page)
    {
        this.page = page;
    }


    public void setTotal_pages (String total_pages)
    {
        this.total_pages = total_pages;
    }


    @Override
    public String toString()
    {
        return "ClassPojo [results = "+results+", page = "+page+", total_pages = "+total_pages+", total_results = "+total_results+"]";
    }
}

File: app/src/main/java/com/pratik/moviesapp/models/Movie.kt
package com.pratik.moviesapp.models

data class Movie(val results: Array<Results>, val page: String, val total_pages: String, val total_results: String) {
    override fun toString(): String {
        return "ClassPojo [results = $results, page = $page, total_pages = $total_pages, total_results = $total_results]"
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Movie

        if (!results.contentEquals(other.results)) return false
        if (page != other.page) return false
        if (total_pages != other.total_pages) return false
        if (total_results != other.total_results) return false

        return true
    }

    override fun hashCode(): Int {
        var result = results.contentHashCode()
        result = 31 * result + page.hashCode()
        result = 31 * result + total_pages.hashCode()
        result = 31 * result + total_results.hashCode()
        return result
    }
}